From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Sat, 8 Jul 2017 02:38:54 -0400
Subject: [PATCH 011/101] disable SLAB_MERGE_DEFAULT by default

---
 mm/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/mm/Kconfig b/mm/Kconfig
index 10057a46bc1b1..613e907d0d514 100644
--- a/mm/Kconfig
+++ b/mm/Kconfig
@@ -185,7 +185,6 @@ config SLUB_TINY
 
 config SLAB_MERGE_DEFAULT
 	bool "Allow slab caches to be merged"
-	default y
 	help
 	  For reduced kernel memory fragmentation, slab caches can be
 	  merged when they share the same size and other characteristics.

