From 9f710c4c97e4d3f082a5fbd16bdeb01f211a97b3 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 27 Jan 2020 18:10:06 +0100
Subject: [PATCH 15/58] init: add CONFIG_BUNKERNEL base config item

---
 init/Kconfig | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/init/Kconfig b/init/Kconfig
index 6da8dc5acd57..eb27e399d7a3 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -192,10 +192,16 @@ config THREAD_INFO_IN_TASK
 	  One subtle change that will be needed is to use try_get_task_stack()
 	  and put_task_stack() in save_thread_stack_tsk() and get_wchan().
 
 menu "General setup"
 
+config BUNKERNEL
+	bool "Tune kernel for interactivity"
+	default y
+	help
+	  Tunes the kernel for responsiveness at the cost of throughput and power usage.
+
 config BROKEN
 	bool
 	help
 	  This option allows you to choose whether you want to try to
 	  compile (and fix) old drivers that haven't been updated to
-- 
2.52.0

