From afca8aa8302df0f0d39a4f4b013249d0280262b0 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Mon, 8 May 2017 12:51:54 -0400
Subject: [PATCH 69/166] enable FORTIFY_SOURCE by default

Signed-off-by: Daniel Micay <danielmicay@gmail.com>
---
 security/Kconfig.hardening | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/Kconfig.hardening b/security/Kconfig.hardening
index 11bd5d6c5f7f..e42e319485ae 100644
--- a/security/Kconfig.hardening
+++ b/security/Kconfig.hardening
@@ -216,10 +216,11 @@ menu "Bounds checking"
 config FORTIFY_SOURCE
 	bool "Harden common str/mem functions against buffer overflows"
 	depends on ARCH_HAS_FORTIFY_SOURCE
 	# https://github.com/llvm/llvm-project/issues/53645
 	depends on !X86_32 || !CC_IS_CLANG || CLANG_VERSION >= 160000
+	default y
 	help
 	  Detect overflows of buffers in common string and memory functions
 	  where the compiler can determine and validate the buffer sizes.
 
 config HARDENED_USERCOPY
-- 
2.52.0

