From f11a6e2204c9bbd1989a23c3cd0ddc057c5448c9 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Mon, 29 May 2017 07:08:42 -0400
Subject: [PATCH 76/162] set LEGACY_VSYSCALL_NONE by default

---
 arch/x86/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index 2cec5a981fa2..c6a0f10ae2ec 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -2217,11 +2217,11 @@ config COMPAT_VDSO
 	  are unlikely to be using a buggy version of glibc.
 
 choice
 	prompt "vsyscall table for legacy applications"
 	depends on X86_64
-	default LEGACY_VSYSCALL_XONLY
+	default LEGACY_VSYSCALL_NONE
 	help
 	  Legacy user code that does not know how to find the vDSO expects
 	  to be able to issue three syscalls by calling fixed addresses in
 	  kernel space. Since this location is not randomized with ASLR,
 	  it can be used to assist security vulnerability exploitation.
-- 
2.52.0

