From 70d59fd1d9863ad4c808a6fe713dedf48096b9c3 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Sat, 27 May 2017 07:29:45 -0400
Subject: [PATCH 80/166] disable PROC_VMCORE by default

---
 fs/proc/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fs/proc/Kconfig b/fs/proc/Kconfig
index 6ae966c561e7..27d78d669f95 100644
--- a/fs/proc/Kconfig
+++ b/fs/proc/Kconfig
@@ -39,11 +39,10 @@ config PROC_KCORE
 	  made using this mechanism.
 
 config PROC_VMCORE
 	bool "/proc/vmcore support"
 	depends on PROC_FS && CRASH_DUMP
-	default y
 	help
 	  Exports the dump image of crashed kernel in ELF format.
 
 config PROC_VMCORE_DEVICE_DUMP
 	bool "Device Hardware/Firmware Log Collection"
-- 
2.52.0

