From 3dfbb5623019ed0b121fa28dd8d9cdf48965088a Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Sun, 7 May 2017 18:28:33 -0400
Subject: [PATCH 86/166] disable COMPAT_BRK by default

---
 mm/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/mm/Kconfig b/mm/Kconfig
index b30b97eea2e6..d0fb9f99673e 100644
--- a/mm/Kconfig
+++ b/mm/Kconfig
@@ -301,11 +301,10 @@ config SHUFFLE_PAGE_ALLOCATOR
 
 	  Say Y if unsure.
 
 config COMPAT_BRK
 	bool "Disable heap randomization"
-	default y
 	help
 	  Randomizing heap placement makes heap exploits harder, but it
 	  also breaks ancient binaries (including anything libc5 based).
 	  This option changes the bootup default to heap randomization
 	  disabled, and can be overridden at runtime by setting
-- 
2.52.0

