From fb66f4617961e960a64a015a51dbfe5971f2fb4c Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Wed, 14 May 2025 21:00:09 +0200
Subject: [PATCH 103/180] kconfig: enable MSEAL_SYSTEM_MAPPINGS by default

Signed-off-by: Levente Polyak <levente@leventepolyak.net>
---
 security/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/Kconfig b/security/Kconfig
index a6335394118d..5ab4df72aedd 100644
--- a/security/Kconfig
+++ b/security/Kconfig
@@ -54,10 +54,11 @@ endchoice
 config MSEAL_SYSTEM_MAPPINGS
 	bool "mseal system mappings"
 	depends on 64BIT
 	depends on ARCH_SUPPORTS_MSEAL_SYSTEM_MAPPINGS
 	depends on !CHECKPOINT_RESTORE
+	default y
 	help
 	  Apply mseal on system mappings.
 	  The system mappings includes vdso, vvar, vvar_vclock,
 	  vectors (arm compat-mode), sigpage (arm compat-mode), uprobes.
 
-- 
2.52.0

