From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Mon, 10 Feb 2026 00:00:00 +0000
Subject: [PATCH 155/180] enable EFI_DISABLE_PCI_DMA by default

Signed-off-by: Daniel Micay <danielmicay@gmail.com>
---
 drivers/firmware/efi/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/firmware/efi/Kconfig b/drivers/firmware/efi/Kconfig
index 29e0729299f5..246bd66d8ec0 100644
--- a/drivers/firmware/efi/Kconfig
+++ b/drivers/firmware/efi/Kconfig
@@ -196,6 +196,7 @@

 config EFI_DISABLE_PCI_DMA
        bool "Clear Busmaster bit on PCI bridges during ExitBootServices()"
+       default y
        help
 	  Disable the busmaster bit in the control register on all PCI bridges
 	  while calling ExitBootServices() and passing control to the runtime
-- 
2.48.1

