From 760a50df9ca99e5665e46f358c9642847cd43fc9 Mon Sep 17 00:00:00 2001
From: Sultan Alsawaf <sultan@kerneltoast.com>
Date: Sat, 28 Mar 2020 13:06:28 -0700
Subject: [PATCH 21/58] mm: disable watermark boosting by default

---
 init/Kconfig    | 1 +
 mm/page_alloc.c | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/init/Kconfig b/init/Kconfig
index 66480d3bc9b1..a95d00951c94 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -199,10 +199,11 @@ config BUNKERNEL
 
 	  --- Virtual Memory Subsystem ---------------------------
 
 	    Background-reclaim hugepages...:   no   ->   yes
 	    Compact unevictable............:   yes  ->   no
+	    Watermark boost factor.........:   1.5  ->   0
 
 	  --- EEVDF CPU Scheduler --------------------------------
 
 	    Minimal granularity............:   0.7  ->   0.4  ms
 	    Migration cost.................:   0.5  ->   0.3  ms
diff --git a/mm/page_alloc.c b/mm/page_alloc.c
index 677f9bb0428b..a5c0bf4cc4c3 100644
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -274,11 +274,15 @@ const char * const migratetype_names[MIGRATE_TYPES] = {
 #endif
 };
 
 int min_free_kbytes = 1024;
 int user_min_free_kbytes = -1;
+#ifdef CONFIG_BUNKERNEL
+static int watermark_boost_factor __read_mostly;
+#else
 static int watermark_boost_factor __read_mostly = 15000;
+#endif
 static int watermark_scale_factor = 10;
 int defrag_mode;
 
 /* movable_zone is the "real" zone pages in ZONE_MOVABLE are taken from */
 int movable_zone;
-- 
2.52.0

