From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Mon, 8 May 2017 12:51:54 -0400
Subject: [PATCH 012/102] enable FORTIFY_SOURCE by default

Signed-off-by: Daniel Micay <danielmicay@gmail.com>
---
 security/Kconfig.hardening | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/Kconfig.hardening b/security/Kconfig.hardening
index ba4e3165d8501..d894025d32c21 100644
--- a/security/Kconfig.hardening
+++ b/security/Kconfig.hardening
@@ -212,10 +212,11 @@ endmenu
 menu "Bounds checking"
 
 config FORTIFY_SOURCE
 	bool "Harden common str/mem functions against buffer overflows"
 	depends on ARCH_HAS_FORTIFY_SOURCE
+	default y
 	help
 	  Detect overflows of buffers in common string and memory functions
 	  where the compiler can determine and validate the buffer sizes.
 
 config HARDENED_USERCOPY
-- 
2.55.0

