From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Wed, 25 Aug 2021 22:24:10 +0200
Subject: [PATCH 044/102] kconfig: enable RANDOMIZE_KSTACK_OFFSET_DEFAULT by
 default

---
 arch/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/Kconfig b/arch/Kconfig
index bf9b5af39caba..f51e0399eadc1 100644
--- a/arch/Kconfig
+++ b/arch/Kconfig
@@ -1562,10 +1562,11 @@ config RANDOMIZE_KSTACK_OFFSET
 	  If unsure, say Y.
 
 config RANDOMIZE_KSTACK_OFFSET_DEFAULT
 	bool "Default state of kernel stack offset randomization"
 	depends on RANDOMIZE_KSTACK_OFFSET
+	default y
 	help
 	  Kernel stack offset randomization is controlled by kernel boot param
 	  "randomize_kstack_offset=on/off", and this config chooses the default
 	  boot state.
 
-- 
2.55.0

