From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Micay <daniel.micay@grapheneos.org>
Date: Thu, 6 Feb 2025 07:32:14 -0500
Subject: [PATCH] zero memory in early boot

---
diff --git a/arch/arm/mm/init.c b/arch/arm/mm/init.c
--- a/arch/arm/mm/init.c
+++ b/arch/arm/mm/init.c
@@ -205,6 +205,8 @@
 
 	find_limits(&min_low_pfn, &max_low_pfn, &max_pfn);
 
+	early_memzero((phys_addr_t)min_low_pfn << PAGE_SHIFT,
+		      (phys_addr_t)max_low_pfn << PAGE_SHIFT);
 	early_memtest((phys_addr_t)min_low_pfn << PAGE_SHIFT,
 		      (phys_addr_t)max_low_pfn << PAGE_SHIFT);
 }
diff --git a/arch/arm64/mm/init.c b/arch/arm64/mm/init.c
--- a/arch/arm64/mm/init.c
+++ b/arch/arm64/mm/init.c
@@ -308,6 +308,7 @@
 	min = PFN_UP(memblock_start_of_DRAM());
 	max = PFN_DOWN(memblock_end_of_DRAM());
 
+	early_memzero(min << PAGE_SHIFT, max << PAGE_SHIFT);
 	early_memtest(min << PAGE_SHIFT, max << PAGE_SHIFT);
 
 	max_pfn = max_low_pfn = max;
diff --git a/arch/x86/mm/init.c b/arch/x86/mm/init.c
--- a/arch/x86/mm/init.c
+++ b/arch/x86/mm/init.c
@@ -809,6 +809,7 @@
 
 	x86_init.hyper.init_mem_mapping();
 
+	early_memzero(0, max_pfn_mapped << PAGE_SHIFT);
 	early_memtest(0, max_pfn_mapped << PAGE_SHIFT);
 }
 
diff --git a/include/linux/memblock.h b/include/linux/memblock.h
--- a/include/linux/memblock.h
+++ b/include/linux/memblock.h
@@ -613,6 +613,8 @@
 static inline void memtest_report_meminfo(struct seq_file *m) { }
 #endif
 
+extern void early_memzero(phys_addr_t start, phys_addr_t end);
+
 #ifdef CONFIG_KEXEC_HANDOVER
 void memblock_set_kho_noprsrv_only(void);
 void memblock_clear_kho_noprsrv_only(void);
diff --git a/mm/Makefile b/mm/Makefile
--- a/mm/Makefile
+++ b/mm/Makefile
@@ -48,7 +48,7 @@
 mmu-$(CONFIG_MMU)	+= mseal.o
 endif
 
-obj-y			:= filemap.o mempool.o oom_kill.o fadvise.o \
+obj-y			:= memzero.o filemap.o mempool.o oom_kill.o fadvise.o \
 			   maccess.o page-writeback.o folio-compat.o \
 			   readahead.o folio.o truncate.o vmscan.o shrinker.o \
 			   shmem.o util.o mmzone.o vmstat.o backing-dev.o \
diff --git a/mm/memzero.c b/mm/memzero.c
--- a/mm/memzero.c
+++ b/mm/memzero.c
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: GPL-2.0
+#include <linux/kernel.h>
+#include <linux/types.h>
+#include <linux/init.h>
+#include <linux/memblock.h>
+#include <linux/string.h>
+
+void __init early_memzero(phys_addr_t start, phys_addr_t end)
+{
+	u64 i;
+	phys_addr_t this_start, this_end;
+
+	pr_info("Early memory zeroing");
+	for_each_free_mem_range(i, NUMA_NO_NODE, MEMBLOCK_NONE, &this_start,
+				&this_end, NULL) {
+		this_start = clamp(this_start, start, end);
+		this_end = clamp(this_end, start, end);
+		if (this_start < this_end) {
+			pr_info("  zero %pa - %pa\n", &this_start, &this_end);
+			memzero_explicit(__va(this_start), this_end - this_start);
+		}
+	}
+}
