From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Sun, 25 Feb 2018 02:15:24 -0500
Subject: [PATCH 036/103] enable AUDIT by default

---
 init/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/init/Kconfig b/init/Kconfig
index c7d2033a68ac1..72c09ecf8e050 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -534,10 +534,11 @@ config CROSS_MEMORY_ATTACH
 	  See the man page for more details.
 
 config AUDIT
 	bool "Auditing support"
 	depends on NET
+	default y
 	help
 	  Enable auditing infrastructure that can be used with another
 	  kernel subsystem, such as SELinux (which requires this for
 	  logging of avc messages output).  System call auditing is included
 	  on architectures which support it.
-- 
2.55.0

