From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Sun, 28 May 2017 18:51:30 -0400
Subject: [PATCH 052/103] mark kmem_cache as __ro_after_init

---
 mm/slab_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/slab_common.c b/mm/slab_common.c
index 0d3d9419bd8e6..a8995782fd7df 100644
--- a/mm/slab_common.c
+++ b/mm/slab_common.c
@@ -38,11 +38,11 @@
 #include <trace/events/kmem.h>
 
 enum slab_state slab_state __ro_after_init;
 LIST_HEAD(slab_caches);
 DEFINE_MUTEX(slab_mutex);
-struct kmem_cache *kmem_cache;
+struct kmem_cache *kmem_cache __ro_after_init;
 
 /*
  * Set of flags that will prevent slab merging.
  * Any flag that adds per-object metadata should be included,
  * since slab merging can update s->inuse that affects the metadata layout.
-- 
2.55.0

