From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Amaan Qureshi <git@amaanq.com>
Subject: [PATCH] mm/slub: add canary on sheaf alloc and free

Signed-off-by: Amaan Qureshi <git@amaanq.com>
---
--- a/mm/slab.h
+++ b/mm/slab.h
@@ -227,6 +227,7 @@
 #ifdef CONFIG_SLAB_CANARY
 	unsigned long random_active;
 	unsigned long random_inactive;
+	unsigned long sheaf_random_active;
 #endif
 
 #ifdef CONFIG_NUMA
--- a/mm/slub.c
+++ b/mm/slub.c
@@ -4905,6 +4905,11 @@
 
 	pcs->main->size--;
 
+	if (!is_kfence_address(object)) {
+		check_canary(s, object, s->random_active);
+		set_canary(s, object, s->sheaf_random_active);
+	}
+
 	local_unlock(&s->cpu_sheaves->lock);
 
 	stat(s, ALLOC_FASTPATH);
@@ -4977,6 +4982,8 @@
 	main->size -= batch;
 	memcpy(p, main->objects + main->size, batch * sizeof(void *));
 
+	check_set_canary_bulk(s, batch, p, s->random_active, s->sheaf_random_active);
+
 	local_unlock(&s->cpu_sheaves->lock);
 
 	stat_add(s, ALLOC_FASTPATH, batch);
@@ -5009,6 +5016,7 @@
 	void *object;
 	bool init = false;
 	bool from_pcs = false;
+	bool from_pcs_failed = false;
 
 	s = slab_pre_alloc_hook(s, gfpflags);
 	if (unlikely(!s))
@@ -5018,13 +5026,16 @@
 	if (unlikely(object))
 		goto out;
 
-	object = alloc_from_pcs(s, gfpflags, node);
-
-	if (object)
+	if (s->cpu_sheaves) {
+		object = alloc_from_pcs(s, gfpflags, node);
 		from_pcs = true;
+	}
 
-	if (!object)
+	if (!object) {
 		object = __slab_alloc_node(s, gfpflags, node, addr, orig_size);
+		if (from_pcs)
+			from_pcs_failed = true;
+	}
 
 	maybe_wipe_obj_freeptr(s, object);
 
@@ -5041,9 +5052,18 @@
 		init = slab_want_init_on_alloc(gfpflags, s);
 	}
 
+	/*
+	 * linux-hardened: In the scenario where an object is intended to be allocated
+	 * from a sheaf but it's allocation failed, it is instead directly allocated from the
+	 * slab allocator but will later be freed back to a sheaf. We thus need to
+	 * set the canary to a sheaf_random_active.
+	 */
 	if (object && !from_pcs) {
 		check_canary(s, object, s->random_inactive);
 		set_canary(s, object, s->random_active);
+	} else if (object && from_pcs_failed) {
+		check_canary(s, object, s->random_inactive);
+		set_canary(s, object, s->sheaf_random_active);
 	}
 
 out:
@@ -5364,6 +5384,11 @@
 
 	/* add __GFP_NOFAIL to force successful memcg charging */
 	slab_post_alloc_hook(s, NULL, gfp | __GFP_NOFAIL, 1, &ret, init, s->object_size);
+
+	if (!is_kfence_address(ret)) {
+		check_canary(s, ret, s->random_active);
+		set_canary(s, ret, s->sheaf_random_active);
+	}
 out:
 	trace_kmem_cache_alloc(_RET_IP_, ret, s, gfp, NUMA_NO_NODE);
 
@@ -5974,6 +5999,10 @@
 			return false;
 	}
 
+	if (!is_kfence_address(object)) {
+		check_canary(s, object, s->sheaf_random_active);
+		set_canary(s, object, s->random_active);
+	}
 	pcs->main->objects[pcs->main->size++] = object;
 
 	local_unlock(&s->cpu_sheaves->lock);
@@ -6126,6 +6155,11 @@
 	 * Since we flush immediately when size reaches capacity, we never reach
 	 * this with size already at capacity, so no OOB write is possible.
 	 */
+
+	if (!is_kfence_address(obj)) {
+		check_canary(s, obj, s->sheaf_random_active);
+		set_canary(s, obj, s->random_active);
+	}
 	rcu_sheaf->objects[rcu_sheaf->size++] = obj;
 
 	if (likely(rcu_sheaf->size < s->sheaf_capacity)) {
@@ -6232,6 +6266,11 @@
 			continue;
 		}
 
+		if (!is_kfence_address(p[i])) {
+			check_canary(s, p[i], s->sheaf_random_active);
+			set_canary(s, p[i], s->random_active);
+		}
+
 		if (unlikely(!can_free_to_pcs(slab))) {
 			remote_objects[remote_nr] = p[i];
 			p[i] = p[--size];
@@ -6405,6 +6444,7 @@
 	       unsigned long addr)
 {
 	bool canary = true;
+	bool to_sheaf = false;
 
 	memcg_slab_free_hook(s, slab, &object, 1);
 	alloc_tagging_slab_free_hook(s, slab, &object, 1);
@@ -6413,7 +6453,7 @@
 	if (is_kfence_address(object))
 		canary = false;
 
-	/* Do not check or set canary if the object is freed back to pcs. */
+	/* Defer canary checking if the object is freed back to pcs. */
 	if (cache_has_sheaves(s) && likely(!IS_ENABLED(CONFIG_NUMA) ||
 				     slab_nid(slab) == numa_mem_id())
 			   && likely(!slab_test_pfmemalloc(slab))) {
@@ -6423,13 +6463,21 @@
 	if (unlikely(!slab_free_hook(s, object, slab_want_init_on_free(s), false, canary)))
 		return;
 
-	if (likely(can_free_to_pcs(slab)) && likely(free_to_pcs(s, object, true)))
-		return;
-
+	if (likely(can_free_to_pcs(slab))) {
+		to_sheaf = true;
+		if (likely(free_to_pcs(s, object, true)))
+			return;
+	}
 
-	/* PCS free failed — transition canary for slab freelist */
-	if (canary && !is_kfence_address(object))
+	/*
+	 * linux-hardened: In this scenario, the object was intended to be freed to a
+	 * sheaf but it failed. The object will thus be freed back to the slab allocator,
+	 * the canary thus need to be checked as a sheaf one and set back to a slab inactive one.
+	 */
+	if (to_sheaf && canary) {
+		check_canary(s, object, s->sheaf_random_active);
 		set_canary(s, object, s->random_inactive);
+	}
 
 	__slab_free(s, slab, object, object, 1, addr);
 	stat(s, FREE_SLOWPATH);
@@ -8789,6 +8837,10 @@
 #ifdef CONFIG_SLAB_CANARY
 	s->random_active = get_random_long();
 	s->random_inactive = get_random_long();
+	if (__slub_debug_enabled())
+		s->sheaf_random_active = s->random_active;
+	else
+		s->sheaf_random_active = get_random_long();
 #endif
 	s->align = args->align;
 	s->ctor = args->ctor;
