From 2ad604a76f523d421b7d061be4e1f2a2ca5d1ef9 Mon Sep 17 00:00:00 2001
From: Amaan Qureshi <git@amaanq.com>
Date: Fri, 13 Feb 2026 23:05:43 -0500
Subject: [PATCH] btf: Verify BTF sorting

---
 kernel/bpf/btf.c | 43 +++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)

diff --git a/kernel/bpf/btf.c b/kernel/bpf/btf.c
index fcd9aabdbd32..eefb73e4ad55 100644
--- a/kernel/bpf/btf.c
+++ b/kernel/bpf/btf.c
@@ -548,10 +548,51 @@ u32 btf_nr_types(const struct btf *btf)
 	}
 
 	return total;
 }
 
+/*
+ * Note that vmlinux and kernel module BTFs are always sorted
+ * during the building phase.
+ */
+static void btf_check_sorted(struct btf *btf)
+{
+	u32 i, n, named_start_id = 0;
+
+	n = btf_nr_types(btf);
+	if (btf_is_vmlinux(btf)) {
+		for (i = btf_start_id(btf); i < n; i++) {
+			const struct btf_type *t = btf_type_by_id(btf, i);
+			const char *n = btf_name_by_offset(btf, t->name_off);
+
+			if (n[0] != '\0') {
+				btf->named_start_id = i;
+				return;
+			}
+		}
+		return;
+	}
+
+	for (i = btf_start_id(btf) + 1; i < n; i++) {
+		const struct btf_type *ta = btf_type_by_id(btf, i - 1);
+		const struct btf_type *tb = btf_type_by_id(btf, i);
+		const char *na = btf_name_by_offset(btf, ta->name_off);
+		const char *nb = btf_name_by_offset(btf, tb->name_off);
+
+		if (strcmp(na, nb) > 0)
+			return;
+
+		if (named_start_id == 0 && na[0] != '\0')
+			named_start_id = i - 1;
+		if (named_start_id == 0 && nb[0] != '\0')
+			named_start_id = i;
+	}
+
+	if (named_start_id)
+		btf->named_start_id = named_start_id;
+}
+
 /*
  * btf_named_start_id - Get the named starting ID for the BTF
  * @btf: Pointer to the target BTF object
  * @own: Flag indicating whether to query only the current BTF (true = current BTF only,
  *       false = recursively traverse the base BTF chain)
@@ -6299,10 +6340,11 @@ static struct btf *btf_parse_base(struct btf_verifier_env *env, const char *name
 
 	err = btf_check_type_tags(env, btf, 1);
 	if (err)
 		goto errout;
 
+	btf_check_sorted(btf);
 	refcount_set(&btf->refcnt, 1);
 
 	return btf;
 
 errout:
@@ -6433,10 +6475,11 @@ static struct btf *btf_parse_module(const char *module_name, const void *data,
 		btf_free(base_btf);
 		base_btf = vmlinux_btf;
 	}
 
 	btf_verifier_env_free(env);
+	btf_check_sorted(btf);
 	refcount_set(&btf->refcnt, 1);
 	return btf;
 
 errout:
 	btf_verifier_env_free(env);
-- 
2.52.0

